Source/GUI/FileInformation.cpp

```diff
@@ -91,182 +92,193 @@ FileInformation::FileInformation (MainWindow* Main_, const QString &FileName_, a
-    {
-        delete Glue;
-        Glue=NULL;
-        for (size_t Pos=0; Pos<Stats.size(); Pos++)
-            Stats[Pos]->StatsFinish();
-    }
+            AudioStats* Audio=new AudioStats();
+            Stats.push_back(Audio);
+            Audio->StatsFromExternalData(StatsFromExternalData);
+        }
+        Glue=new FFmpeg_Glue(FileName_string.c_str(), ActiveAllTracks, &Stats, Stats.empty());
+        if (FileName_string.empty())
+        {
+            Glue->AddInput_Video(FrameCount, 1001, 30000, 720, 486, blackmagicDeckLink_Glue->Config_In.VideoBitDepth, blackmagicDeckLink_Glue->Config_In.VideoCompression, blackmagicDeckLink_Glue->Config_In.TC_in);
+            Glue->AddInput_Audio(FrameCount, 1001, 30000, 48000, blackmagicDeckLink_Glue->Config_In.AudioBitDepth, blackmagicDeckLink_Glue->Config_In.AudioTargetBitDepth, blackmagicDeckLink_Glue->Config_In.ChannelsCount);
+        }
+        else if (Glue->ContainerFormat_Get().empty())
+        {
+            delete Glue;
+            Glue=NULL;
+            for (size_t Pos=0; Pos<Stats.size(); Pos++)
+                Stats[Pos]->StatsFinish();
+        }
 
-    if (Glue)
-    {
-        Glue->AddOutput(0, 72, 72, FFmpeg_Glue::Output_Jpeg);
-        if (!Encoding_FileName.empty())
+        if (Glue)
         {
-            Glue->AddOutput(Encoding_FileName, Encoding_Format);
-            FileName=QString::fromUtf8(Encoding_FileName.c_str());
+            Glue->AddOutput(0, 72, 72, FFmpeg_Glue::Output_Jpeg);
+            if (!Encoding_FileName.empty())
+            {
+                Glue->AddOutput(Encoding_FileName, Encoding_Format);
+                FileName=QString::fromUtf8(Encoding_FileName.c_str());
+            }
+            Glue->AddOutput(1, 0, 0, FFmpeg_Glue::Output_Stats, 0, Filters[0]);
+            Glue->AddOutput(0, 0, 0, FFmpeg_Glue::Output_Stats, 1, Filters[1]);
         }
-        Glue->AddOutput(1, 0, 0, FFmpeg_Glue::Output_Stats, 0, Filters[0]);
-        Glue->AddOutput(0, 0, 0, FFmpeg_Glue::Output_Stats, 1, Filters[1]);
-    }
 
-    // Looking for the first video stream
-    ReferenceStream_Pos=0;
-    for (; ReferenceStream_Pos<Stats.size(); ReferenceStream_Pos++)
-        if (Stats[ReferenceStream_Pos] && Stats[ReferenceStream_Pos]->Type_Get()==0)
-            break;
-    if (ReferenceStream_Pos>=Stats.size())
+        // Looking for the first video stream
         ReferenceStream_Pos=0;
+        for (; ReferenceStream_Pos<Stats.size(); ReferenceStream_Pos++)
+            if (Stats[ReferenceStream_Pos] && Stats[ReferenceStream_Pos]->Type_Get()==0)
+                break;
+        if (ReferenceStream_Pos>=Stats.size())
+            ReferenceStream_Pos=0;
 
-    Frames_Pos=0;
-    WantToStop=false;
-    Parse();
+        Frames_Pos=0;
+        WantToStop=false;
+        Parse();
+
+    }
+
+    catch(const std::bad_alloc&)
+    {
+        QMessageBox::critical(NULL, "Out of memory!", "Please try 64bit build.");
+        QApplication::instance()->quit();
```

Comment:
Just saying that if the user provides an empty XML, you'll get the same behavior that what you want to avoid, so replacing a bad_alloc by an empty XML is not worse than current behavior with empty XML.